conda_index/index/__init__.py

```diff
@@ -748,6 +752,11 @@ def index_subdir(self, subdir, verbose=False, progress=False):
             "removed": [],  # can be added by patch/hotfix process
         }
 
+        if self.base_url:
+            # per https://github.com/conda-incubator/ceps/blob/main/cep-15.md
+            new_repodata["info"]["base_url"] = f"{self.base_url.rstrip('/')}/{subdir}/"
+            new_repodata["repodata_version"] = 2
```

Comment:
Assuming you'd be okay to add a --repodata-version option (which defaults to 1), we could add the repodata_version to all repodata generated?

```suggestion
        new_repodata["repodata_version"] = self.repodata_version
        if self.repodata_version == 2:
            if not self.base_url:
                raise ValueError(
                    "Repodata version 2 was request but no base URL specified."
                )
            # per https://github.com/conda-incubator/ceps/blob/main/cep-15.md
            new_repodata["info"]["base_url"] = f"{self.base_url.rstrip('/')}/{subdir}/"
```

Reply:
I was thinking along the lines of automatically choosing "the smallest repodata version number supporting features requested"